driver/src/transport.rs

```diff
@@ -1,79 +1,123 @@
-use ethcontract::jsonrpc::types::request::Call;
-use ethcontract::web3::error::Error;
-use ethcontract::web3::futures::{Async, Future, Poll};
-use ethcontract::web3::{RequestId, Transport};
-use log::{log, Level};
+use crate::error::DriverError;
+use ethcontract::jsonrpc::types::{Call, Output};
+use ethcontract::web3::helpers;
+use ethcontract::web3::{Error as Web3Error, RequestId, Transport};
+use futures::compat::Compat;
+use futures::future::{BoxFuture, FutureExt, TryFutureExt};
+use isahc::config::VersionNegotiation;
+use isahc::prelude::{Body, Request, Response};
+use isahc::{Error as HttpError, HttpClient, ResponseExt};
+use log::{debug, warn};
+use serde::Deserialize;
 use serde_json::Value;
+use std::fmt::{self, Debug, Formatter};
+use std::str::FromStr;
+use std::sync::atomic::{AtomicUsize, Ordering};
+use std::sync::Arc;
+use std::time::Duration;
```

Comment:
Seems we are "using" a lot of stuff. Is there a more compact way of doing this? Are we using all of these imports or are they somehow being used indirectly here? Seems like there aren't even enough lines in this file to use all this stuff!

Reply:
It should be a warning and thus an error because we don't allow warnings when there are unused imports.

Reply:
Its all used. Like @e00E said unused imports fail the build. Its just this transport is "gluing" a lot of separate pieces together, so it end up importing a lot.